TUTORIAL.md

```diff
@@ -307,16 +307,34 @@ previous_response
 # Same as paginated_response_page_1
 ```
 
-And `.iterator()` which returns iterable enabling looping through the responses in for loop:
+And `.iterator` which returns iterable enabling looping through the rest of responses in for loop:
 
 ```python
+# first response page
 paginated_response_page_1 = session.flaws.retrieve_list()
 
-for response in paginated_response_page_1.iterator():
-    # do stuff with response
+# iterator starting with second page
+for response in paginated_response_page_1.iterator:
+    # do stuff with second, third, fourth, ... response page
+    print(response)
```

Comment:
Two things to remark here:

I think an example like this is clearer:

```python
paginated_response = session.flaws.retrieve_list()
for page in paginated_response.iterator:
    for flaw in page:
        do_calc(flaw)
```

And the other thing is that there doesn't seem to be a way to apply a function generically to every page? Correct me if I'm wrong but if I wan't to apply some function to every single page I'd have to do the following:

```python
first_page = session.flaws.retrieve_list()
do_calc(first_page)
for page in first_page.iterator:
    do_calc(page)
```

I think that the better API would be something like:

```python
for page in sessions.flaws.retrieve_list_iterator():
    for flaw in page:
        do_calc(flaw)
```

or ideally:

```python
for flaw in session.flaws.retrieve_list():
    do_calc(flaw)
```

for the latter retrieve_list() would have to return a generator of flaw objects that yields from a generator of pages

Reply:
Yeah that's fine, you can do it in a myriad of ways but probably the most backwards-compatible one would be either a different method (e.g. `retrieve_list_iterator()` or have an argument like `retrieve_list(auto_paginate=True)`